service/matching/task_queue_partition_manager.go

```diff
@@ -257,6 +259,14 @@ func (pm *taskQueuePartitionManagerImpl) AddTask(
 	return assignedBuildId, false, spoolQueue.SpoolTask(params.taskInfo)
 }
 
+func (pm *taskQueuePartitionManagerImpl) shouldBacklogSyncMatchTaskOnError(err error) bool {
+	if err != nil && errors.Is(err, consts.ErrResourceExhaustedBusyWorkflow) {
```

Comment:
We should consider extend to other error types as well, like timeout. 
@ShahabT is the main concern here that by adding the task to backlog matching will keep retrying the task and make it a busy loop?

Reply:
Yes, I don't think we should add error types that'll fail immediately when retried. Timeout seems safe in that sense. Don't remember seeing it though. maybe we can add a task for it next time we see the error?